File: ISS Emergency/Assets/Scripts/TriggerPanelButton.js
#pragma strict

// Hannes Helmholz
//
// 


private class TriggerPanelButton extends MonoBehaviour {

	// =============================================================================
	// MEMBERS ---------------------------------------------------------------------
	@SerializeField private var ColorOn:Color;
	
	private var Mat:Material;
	private var Audio:AudioSource;
	private var State:boolean = false;
	private var InitState:boolean = true;
	private var ColorOff:Color;
	// ============================================================================
	
	
	
	// =============================================================================
	// METHODS UNITY ---------------------------------------------------------------
	
	function Awake() {
		Mat = renderer.material;
		Audio = audio;
		
		ColorOff = Mat.color;
	}
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS ---------------------------------------------------------------------
	
	public function SetState(state:boolean) {
		if (state != State)
			Trigger();
	}
	
	
	public function Trigger() {
		State = !State;
		
		if (InitState) {
			//Mat.color = State ? ColorOn : ColorOff;
			Mat.color = ColorOn;
			InitState = false;
		}
		
		if (Audio) {
			if (State)
				Audio.Play();
			else
				Audio.Stop();
		}
	}
	// =============================================================================

}

File: Assets/Scripts/TriggerPanelButton.cs

// Hannes Helmholz
// Diese Klasse lösst den Sirenen Sound aus, der beim aktivieren eines Panels aktiviert wird. Desweiterin
// wird das blinken der Panels hier ausgelöst.
// 

using UnityEngine;

public class TriggerPanelButton : MonoBehaviour
{

    // =============================================================================
    // MEMBERS ---------------------------------------------------------------------
    [SerializeField]
    private Color ColorOn;
    private Material Mat;
    private AudioSource Audio;
    private bool State = false;
    private bool InitState = true;
    private Color ColorOff;
    // ============================================================================
 
 
 
    // =============================================================================
    // METHODS UNITY ---------------------------------------------------------------
 
    protected virtual void Awake ()
    {
        Mat = renderer.material;
        Audio = audio;
     
        ColorOff = Mat.color;
    }
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS ---------------------------------------------------------------------
 
    public void SetState ( bool state )
    {
        if ( state != State )
        {
            Trigger ();
        }
    }
 
 
    public void Trigger ()
    {
        State = !State;
     
        if ( InitState )
        {
            //Mat.color = State ? ColorOn : ColorOff;
            Mat.color = ColorOn;
            InitState = false;
        }
     
        if ( Audio )
        {
            if ( State )
            {
                Audio.Play ();
            }
            else
            {
                Audio.Stop ();
            }
        }
    }
    // =============================================================================

}